pkg/controller/acmeorders/sync.go

```diff
@@ -452,10 +452,14 @@ func challengeSpecForAuthorization(ctx context.Context, cl acmecl.Interface, iss
 			continue
 		}
 
-		if cfg.Selector == nil && selectedSolver == nil {
-			dbg.Info("selecting solver due to nil selector and no previously selected solver")
-			selectedSolver = cfg.DeepCopy()
-			selectedChallenge = acmech
+		if cfg.Selector == nil {
+			if selectedSolver == nil {
+				dbg.Info("selecting solver due to nil selector and no previously selected solver")
+				selectedSolver = cfg.DeepCopy()
+				selectedChallenge = acmech
+			} else {
+				dbg.Info("not selecting solver as previously selected solver has a just as or more specific selector")
+			}
```

Comment:
nit: in go, using `if { .. } else { .. }` tends to be avoided and rewritten as:

```go
		if cfg.Selector == nil {
			if selectedSolver != nil {
				dbg.Info("not selecting solver as previously selected solver has a just as or more specific selector")
				continue
			}
			dbg.Info("selecting solver due to match all selector and no previously selected solver")
			selectedSolver = cfg.DeepCopy()
			selectedChallenge = acmech
			continue
		}
```

(also note I updated the debug message in that example to change `nil`->`match all` to make it clearer)

Reply:
Made the change :+1: